centreon-broker/core/inc/com/centreon/broker/multiplexing/muxer.hh

```diff
@@ -67,7 +72,21 @@ class muxer : public io::stream {
   void _get_event_from_file(std::shared_ptr<io::data>& event);
   std::string _memory_file() const;
   void _push_to_queue(std::shared_ptr<io::data> const& event);
-  std::string _queue_file() const;
+
+  void updateStats(void) noexcept {
+    auto now(std::chrono::system_clock::now());
+    if (std::chrono::duration_cast<std::chrono::milliseconds>(now - _clk)
+            .count() > 1000) {
+      _clk = now;
+      _stats->set_queue_file_enabled(_queue_file_enabled);
+      _stats->set_queue_file(_queue_file_name);
+      _stats->set_unacknowledged_events(std::to_string(_events_size));
```

Comment:
unacknowledged events have no relation to _events_size. Here we talk about data already sent to the stream but not acknowledged.

Reply:
There is a concurrent access on _stats.
If someone asks for statistics while you write them...